tagcompare/logger.py

```diff
@@ -0,0 +1,56 @@
+import os
+import logging
+
+import settings
+
+
+class Logger(object):
+    def __init__(self, name, directory=None, writefile=False):
+        if directory and not writefile:
+            raise ValueError("directory was specified when writefile is False!")
+
+        self.__directory = directory
+        if writefile:
+            if not directory:
+                self.__directory = os.path.join(
+                    os.path.abspath(os.path.dirname(__file__)), settings.LOG_DIR)
+            if not os.path.exists(self.__directory):
+                raise IOError("directory doesn't exist at {}".format(self.__directory))
+
+        self.__name = name.replace('.log', '')
+        logger = logging.getLogger('tagcompare.%s' % name)
+
+        # Set default log levels
+        logging.getLogger("tagcompare").setLevel(logging.DEBUG)
+        if not logger.handlers:
+            if writefile:
+                logger.addHandler(self.__file_handler())
+            logger.addHandler(self.__stream_handler())
+        self._logger = logger
+
+    @property
+    def name(self):
+        return self.__name
+
+    @property
+    def filepath(self):
+        if not self.__directory:
+            raise ValueError('Cannot get filepath when there is no directory set!')
+        return os.path.join(self.__directory, '%s.log' % self.__name)
+
+    def __file_handler(self):
+        handler = logging.FileHandler(self.filepath, mode='a')
+        formatter = logging.Formatter('%(asctime)s %(levelname)s:%(name)s %(message)s')
+        handler.setFormatter(formatter)
+        handler.setLevel(logging.DEBUG)
+        return handler
+
+    def __stream_handler(self):
+        handler = logging.StreamHandler()
+        formatter = logging.Formatter('%(levelname)s:%(name)s %(message)s')
```

Comment:
It really starts to look very messy on the console with timestamp, I actually originally wanted to show only the %(message)s